dali/test/python/test_operator_gridmask.py

```diff
@@ -0,0 +1,103 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import numpy as np
+import math
+import os
+import cv2
+from test_utils import get_dali_extra_path
+
+data_root = get_dali_extra_path()
+img_dir = os.path.join(data_root, 'db', 'single', 'jpeg')
+
+def get_pipeline(batch_size, tile, ratio, angle):
+  pipe = Pipeline(batch_size, 4, None)
+  with pipe:
+    input, _ = fn.file_reader(file_root=img_dir)
+    decoded = fn.image_decoder(input, device='cpu', output_type=types.RGB)
+    grided = fn.grid_mask(decoded, device='cpu', tile=tile, ratio=ratio, angle=angle)
+    pipe.set_outputs(grided, decoded)
+  return pipe
+
+def get_random_pipeline(batch_size):
+  pipe = Pipeline(batch_size, 4, None)
+  with pipe:
+    input, _ = fn.file_reader(file_root=img_dir)
+    decoded = fn.image_decoder(input, device='cpu', output_type=types.RGB)
+    tile = fn.cast(fn.uniform(range=(50, 200), shape=[1]), dtype=types.INT32)
+    ratio = fn.uniform(range=(0.3, 0.7), shape=[1])
+    angle = fn.uniform(range=(-math.pi, math.pi), shape=[1])
+    grided = fn.grid_mask(decoded, device='cpu', tile=tile, ratio=ratio, angle=angle)
+    pipe.set_outputs(grided, decoded, tile, ratio, angle)
+  return pipe
+
+def get_mask(w, h, tile, ratio, angle, d):
+  ca = math.cos(angle)
+  sa = math.sin(angle)
+  b = tile * ratio
+
+  i = np.tile(np.arange(w), (h, 1))
+  j = np.transpose(np.tile(np.arange(h), (w, 1)))
+  x = i * ca - j * sa
+  y = i * sa + j * ca
+  m = np.logical_or(((x+d) % tile > b+2*d), ((y+d) % tile > b+2*d))
+  return m
+
+def check(result, input, tile, ratio, angle):
+  result = np.uint8(result)
+  input = np.uint8(input)
+  w = result.shape[1]
+  h = result.shape[0]
+  eps = 0.1
+  
+  # inside of squares should be black
+  mask = np.uint8(1 - get_mask(w, h, tile, ratio, angle, -eps))
+  result2 = cv2.bitwise_and(result, result, mask=mask)
+  assert not np.any(result2)
+
+  # outside of squares should be same as input
+  mask = np.uint8(get_mask(w, h, tile, ratio, angle, eps))
+  result2 = cv2.bitwise_and(result, result, mask=mask)
+  input2 = cv2.bitwise_and(input, input, mask=mask)
+  assert np.all(result2 == input2)
+
+def test_cpu_vs_cv():
+  batch_size = 4
+  for (tile, ratio, angle) in [(40, 0.5, 0),
+                               (100, 0.1, math.pi / 2),
+                               (200, 0.7, math.pi / 3),
+                               (150, 1/3, math.pi / 4),
+                               (50, 0.532, 1),
+                               (51, 0.38158387, 2.6810782),
+                               (123, 0.456, 0.789)]:
+    pipe = get_pipeline(batch_size, tile, ratio, angle)
+    pipe.build()
+    results, inputs = pipe.run()
+    for i in range(batch_size):
+      yield check, results[i], inputs[i], tile, ratio, angle
+
+def test_cpu_vs_cv_random():
+  batch_size = 4
```

Comment:
You mentioned in the comment that now the tests are slower. I see that the batch size and number of iterations were doubled, which means the test has 4 times more work to do. If you want to reduce testing time I'd reduce the number of iterations (3 should be enough, I think)

Reply:
I did, but then i made them fast again (7f268fd), so I brought back the old parameters. I think the time taken by both tests is fine.

Reply:
It takes 12s, looks good.